treeherder/webapp/api/performance_data.py

```diff
@@ -911,48 +915,42 @@ def list(self, request):
 
         return Response(data=serialized_data)
 
-    def _get_ph_time_ranges(self):
-        return [
-            {'value': 86400, 'text': 'Last day'},
-            {'value': 86400 * 2, 'text': 'Last 2 days'},
-            {'value': 604800, 'text': 'Last 7 days'},
-            {'value': 1209600, 'text': 'Last 14 days'},
-            {'value': 2592000, 'text': 'Last 30 days'},
-            {'value': 5184000, 'text': 'Last 60 days'},
-            {'value': 7776000, 'text': 'Last 90 days'},
-            {'value': 31536000, 'text': 'Last year'},
-        ]
-
-    def _get_push_timestamp(self, base_push, new_push):
+    @staticmethod
+    def _get_push_timestamp(base_push, new_push):
         # This function will determine the right push time stamp to assign a revision.
         # It will do this by comparing timestamps with ph_time_ranges
-        base_push_timestamp = base_push.time
         new_push_timestamp = new_push.time
 
-        timestamps = [base_push_timestamp, new_push_timestamp]
+        timestamps = [new_push_timestamp]
+        if base_push:
+            base_push_timestamp = base_push.time
+            timestamps.append(base_push_timestamp)
 
-        ph_ranges = self._get_ph_time_ranges()
+        timeranges = perfcompare_utils.PERFHERDER_TIMERANGES
         values = []
 
         date_now = (time.time() * 1000) / 1000.0
         for ts in timestamps:
             ph_value = date_now - to_timestamp(str(ts))
-            for ph_range in ph_ranges:
+            for ph_range in timeranges:
                 if ph_value < ph_range['value']:
                     values.append(ph_range['value'])
                     break
         return max(values)
 
-    def _get_perf_data(self, repository_name, revision, signatures, interval):
+    def _get_perf_data(self, repository_name, revision, signatures, interval, startday, endday):
         perf_data = self._get_perf_data_by_repo_and_signatures(repository_name, signatures)
         if revision:
             perf_data = perf_data.filter(push__revision=revision)
-        elif interval:
+        elif interval and not startday and not endday:
             perf_data = perf_data.filter(
                 push_timestamp__gt=datetime.datetime.utcfromtimestamp(
                     int(time.time() - int(interval))
                 )
             )
+        else:
+            perf_data = perf_data.filter(push_timestamp__gt=startday, push_timestamp__lt=endday)
```

Comment:
If the revision is not provided, but startday and endday are, we filter by them to get the performance data.